src/redux/books/booksSlice.js

```diff
@@ -1,26 +1,67 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const URLGETBOOKS = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/wn1qJHo9MjP5T7pAjyNj/books';
+
+// Getting List of Books
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios.get(URLGETBOOKS);
+    const data = await response.data;
+    console.log(data);
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+
+// Adding a Book
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  try {
+    const response = await axios.post(URLGETBOOKS, book);
+    const data = await response.data;
+    alert(data);
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
+
+// Deleting a Book
+export const deleteBook = createAsyncThunk('books/deleteBook', async (id) => {
+  try {
+    const response = await axios.delete(`${URLGETBOOKS}/${id}`);
+    const data = await response.data;
+    alert(data);
+    return data;
+  } catch (error) {
+    return error;
+  }
+});
 
 const initialState = {
   bookItems: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
+    // {
+    //   item_id: 'item1',
+    //   title: 'The Great Gatsby',
+    //   author: 'John Smith',
+    //   category: 'Fiction',
+    // },
+    // {
+    //   item_id: 'item2',
+    //   title: 'Anna Karenina',
+    //   author: 'Leo Tolstoy',
+    //   category: 'Fiction',
+    // },
+    // {
+    //   item_id: 'item3',
+    //   title: 'The Selfish Gene',
+    //   author: 'Richard Dawkins',
+    //   category: 'Nonfiction',
+    // },
```

Comment:
Hi Bro I removed the outdated code and commented code; from lifting state up to Local state to redux state to Now Api Via Redux! It was all Because of Many Refactors